demos/twa-firebase-analytics/src/main/java/com/google/androidbrowserhelper/demos/twa_firebase_analytics/FirebaseAnalyticsLauncherActivity.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.androidbrowserhelper.demos.twa_firebase_analytics;
+
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.androidbrowserhelper.trusted.LauncherActivity;
+import com.google.firebase.analytics.FirebaseAnalytics;
+
+public class FirebaseAnalyticsLauncherActivity extends LauncherActivity {
+    private String mAppInstanceId;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.mLaunchImmediately = false;
+        super.onCreate(savedInstanceState);
+
+        FirebaseAnalytics mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);
+
+        // Start the asynchronous task to get the Firebase application instance id.
+        mFirebaseAnalytics.getAppInstanceId().addOnCompleteListener(new OnCompleteListener<String>() {
```

Comment:
You can use a lambda here:

```
...addOnCompleteListener(task -> {
  // Do stuff with task.
});
```

(you may need to enable Java 8 in gradle, but we use lambdas elsewhere)

Reply:
Done.